### PR TITLE
Add Jetpack CI

### PR DESCRIPTION
**Pull request by jkmassel**
Adds CI for the Jetpack app

To test:
- Ensure that all CI jobs pass
- Ensure that [the installable build shows up in App Center](https://appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/distribute/releases) and is installable on your device

Once this PR is approved, I'll remove the installable build from PRs and only run it nightly. Jetpack builds will continue on each PR.

## Regression Notes
1. Potential unintended areas of impact: n/a
2. What I did to test those areas of impact (or what existing automated tests I relied on): n/a
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/WordPress-Alpha.entitlements [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	</array>
 	<key>com.apple.security.application-groups</key>
 	<array>
-		<string>group.org.wordpress.alpha</string>
+		<string>group.com.jetpack.alpha</string>
```
Comment (leandroalonso) [inline WordPress/WordPress-Alpha.entitlements]: This is changing the `WordPress-Alpha` entitlement. Is that an unintentional change?

File: WordPress/WordPress-Internal.entitlements [hunk 2]
```diff
@@ -3,7 +3,7 @@
 <plist version="1.0">
 <dict>
 	<key>aps-environment</key>
-	<string>distribution</string>
+	<string>development</string>
```
Comment (leandroalonso) [inline WordPress/WordPress-Internal.entitlements]: Same as `WordPress-Alpha`

**Review by leandroalonso — COMMENTED** [review]
@jkmassel everything is 🟢!

On AppCenter I see a WP icon but the app installed is Jetpack, so I'm not sure if this is something on their side.

Also left two comments on changes on files that I think were committed by mistake.

File: WordPress/Resources/AppImages.xcassets/AppIcon-Internal.appiconset/Contents.json [hunk 3]
```diff
@@ -1,42 +1,43 @@
 {
   "images" : [
     {
-      "filename" : "app-icon-40pt.png",
+      "filename" : "Icon-App-20x20@2x-1.png",
```
Comment (leandroalonso) [inline WordPress/Resources/AppImages.xcassets/AppIcon-Internal.appiconset/Contents.json]: I wonder why this was changed too (given it's WP) 🤔

File: WordPress/WordPress-Alpha.entitlements (hunk 1 above)
Comment (jkmassel) [reply WordPress/WordPress-Alpha.entitlements]: That's odd – shouldn't have made it in. I deleted it, thanks!

File: WordPress/WordPress-Internal.entitlements (hunk 2 above)
Comment (jkmassel) [reply WordPress/WordPress-Internal.entitlements]: Same as above

File: WordPress/Resources/AppImages.xcassets/AppIcon-Internal.appiconset/Contents.json (hunk 3 above)
Comment (jkmassel) [reply WordPress/Resources/AppImages.xcassets/AppIcon-Internal.appiconset/Contents.json]: This was from a change I made in a release branch – something must've gone weird in a merge, sorry about that. It's been removed.

**Review by leandroalonso — APPROVED** [review]
:shipit: 

Thank you so much, @jkmassel!

**Pull request merged**